pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -3847,18 +3848,146 @@ func TestChangefeedBareAvro(t *testing.T) {
 	cdcTest(t, testFn, feedTestForceSink("kafka"))
 }
 
+func mustT[T any](t *testing.T) func(x T, err error) T {
+	t.Helper()
+	return func(x T, err error) T {
+		t.Helper()
+		if err != nil {
+			t.Fatal(err)
+		}
+		return x
+	}
+}
+
+func toJSON(t *testing.T, x any) string {
+	t.Helper()
+	// Our json library formats differently than the stdlib, and json.MakeJSON()
+	// chokes on some inputs, so marshal it twice.
+	bs, err := gojson.Marshal(x)
+	require.NoError(t, err)
+	j, err := json.ParseJSON(string(bs))
+	require.NoError(t, err)
+	return j.String()
+}
+
+func parseJSON(t *testing.T, s string) map[string]any {
+	t.Helper()
+	var m map[string]any
+	require.NoError(t, gojson.Unmarshal([]byte(s), &m))
+	return m
+}
+
 func TestChangefeedEnriched(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	key := map[string]any{"a": 0}
+	keySchema := map[string]any{
+		"field": "foo_key",
+		"fields": []map[string]any{
+			{"field": "a", "optional": true, "type": "int64"},
+		},
+		"type": "struct",
+	}
+	afterVal := map[string]any{"a": 0, "b": "dog"}
+	afterSchema := map[string]any{
+		"field": "foo_after",
+		"fields": []map[string]any{
+			{"field": "a", "optional": true, "type": "int64"},
+			{"field": "b", "optional": true, "type": "string"},
+		},
+		"optional": true,
+		"type":     "struct",
+	}
+	sourceSchema := map[string]any{
+		"field": "source",
+		"fields": []map[string]any{
+			{"field": "job_id", "optional": true, "type": "string"},
+		},
+		"optional": true,
+		"type":     "struct",
+	}
+
 	cases := []struct {
 		name               string
 		enrichedProperties []string
+		msgBodyJSONFunc    func(source map[string]any) map[string]any
```

Comment:
nit: could you rename this to something like `makeExpectedJSONBody` to make it more clear this is generating an expected value